src/lib/QueryType.hs

```diff
@@ -471,9 +458,8 @@ dictExprType e = case e of
     liftM DictTy $ ixDictType $ TC $ Fin n'
   ExplicitMethods v params -> do
     params' <- mapM substM params
-    SpecializedDictBinding (SpecializedDictDef (Abs bs dictTy) _) <- lookupEnv =<< substM v
-    dictTy' <- applySubst (bs @@> map SubstVal params') dictTy
-    return $ DictTy dictTy'
+    SpecializedDictBinding (SpecializedDict (Abs bs dict) _) <- lookupEnv =<< substM v
+    getType =<< applySubst (bs @@> map SubstVal params') dict
```

Comment:
It would likely be better to use `getTypeSubst` since that avoids doing the full substitution of the dict.